libs/model/src/policies/ContestWorker.policy.ts

```diff
@@ -1,18 +1,77 @@
 import { schemas, type Policy } from '@hicommonwealth/core';
+import { commonProtocol, models } from '@hicommonwealth/model';
 
 const inputs = {
   ThreadCreated: schemas.events.ThreadCreated,
+  ThreadUpvoted: schemas.events.ThreadUpvoted,
   CommentCreated: schemas.events.CommentCreated,
 };
 
 export const ContestWorker: Policy<typeof inputs> = () => ({
   inputs,
   body: {
     ThreadCreated: async ({ name, payload }) => {
-      console.log(name, payload.thread);
+      console.log(name, payload);
+      const { threadId, userAddress } = payload;
+
+      const contestAddress = '0x123'; // TODO: get contest address from contest metadata projection
+      const contentUrl = `https://`; // TODO: generate content URL programmatically?
```

Comment:
we should have utils to generate content urls.... probably inside the /client space? Should be reusable by client/server.

Reply:
I've created a reusable func for building the URL in core.

Currently, it's a non-trivial effort make all content URLs across the app use a single reusable function:
- The frontend router has the routes hardcoded and some places in the app have hardcoded URLs. That's fine.
- But at the same time, there's the `getProposalUrlPath` func which is used in some parts of the frontend, but it's not a pure function– it relies on the `app` object for custom domain and community ID.
- Additionally, the `useCommonNavigate` function has different call patterns for navigation

In total, there are dozens of places to update and it's not a simple let-the-editor-refactor-it situation.

I recommend we create a separate ticket to make the single-reusable-func change.